engine/pongscenes.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include "gameobject.h"
+#include "scene.h"
+
+#include <memory>
+#include <vector>
+
+namespace pong
+{
+
+class TitleScene : public Scene
+{
+public:
+    void BuildScene() override;
```

Comment:
Curious if you have any thoughts on how I switched up the scene creation. I think it saves from some copy pasting, but still kind of follows the same idea of building a vector of objects. 

A proper way this is handled in game engines is with .scene asset. These consist of a lot of object data (json/yaml) that define the layout of a scene. But I think that will be another challenge converting text into specific component types 😬.

Reply:
I think this is good. I think this is basically the factory design pattern, you could look into that if you wanted some inspiration on a design like this

https://refactoring.guru/design-patterns/factory-method
https://refactoring.guru/design-patterns/abstract-factory